src/evict/evict_lru.c

```diff
@@ -2428,6 +2433,13 @@ __wt_cache_eviction_worker(WT_SESSION_IMPL *session, bool busy, bool readonly, d
         if (__wt_op_timer_fired(session))
             break;
 
+        /* Check if we have exceeded the global or the session timeout for waiting on the cache. */
+        if (time_start != 0 && cache_max_wait_us != 0) {
```

Comment:
To maintain the code consistency for the change in 5.0, this should be:
```if (app_thread && cache_max_wait_us != 0)```

Reply:
[236e0f1](https://github.com/wiredtiger/wiredtiger/pull/9758/commits/236e0f10b76a735a1f3028ed204d7a4e212f6477)